geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/service/GeoMesaDataStoreService.scala

```diff
@@ -39,19 +39,23 @@ class GeoMesaDataStoreService[T <: DataStoreFactorySpi: ClassTag](descriptors: S
 
   override def loadDataStore: DataStore = synchronized {
     if (store == null) {
-      store = tryGetDataStore(params).get
+      store = newDataStore()
     }
     store
   }
 
+  override def newDataStore(): DataStore = synchronized {
+    tryGetDataStore(params).get
+  }
+
   override def dispose(ds: DataStore): Unit = synchronized {
     if (ds != null && !ds.eq(store)) {
       CloseWithLogging(ds)
     }
   }
 
   @OnEnabled
-  def onEnabled(context: ConfigurationContext): Unit = {
+  def onEnabled(context: ConfigurationContext): Unit = synchronized {
```

Comment:
don't think this needs to be synchronized either, the nifi lifecycle ensures that it gets called in the correct order